compiler/rustc_lint_defs/src/builtin.rs

```diff
@@ -1707,6 +1708,33 @@ declare_lint! {
     "detects lifetime parameters that are never used"
 }
 
+declare_lint! {
+    /// The `redundant_lifetimes` lint detects lifetime parameters that are never
+    /// used, or are redundant because they are equal to another named lifetime.
```

Comment:
```suggestion
    /// The `redundant_lifetimes` lint detects lifetime parameters that are
    /// redundant because they are equal to another named lifetime.
```